platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationConfigInstaller.java

```diff
@@ -83,58 +82,63 @@ public void run() {
       return;
     }
-    InputStream is = null;
-    try {
-      is = new FileInputStream(configFile);
+
+    try (InputStream is = new FileInputStream(configFile)) {
       Properties props = new Properties();
       props.load(is);
-      if (!props.isEmpty()) {
-        LOGGER.debug("Found applications to install from config.");
-        for (Entry<Object, Object> curApp : props.entrySet()) {
-          String appName = curApp.getKey().toString();
-          String appLocation = curApp.getValue().toString();
-          LOGGER.debug("Starting app {} at location: {}", appName, appLocation);
-
-          try {
-            if (StringUtils.isNotEmpty(appLocation)) {
-              appService.addApplication(new URI(appLocation));
-            }
-            executeAsSystem(
-                () -> {
-                  appService.startApplication(appName);
-                  return true;
-                });
-          } catch (ApplicationServiceException ase) {
-            LOGGER.warn("Could not start " + appName, ase);
-          } catch (URISyntaxException use) {
-            LOGGER.warn(
-                "Could not install application, location is not a valid URI " + appLocation, use);
-          }
-        }
-        LOGGER.debug("Finished installing applications, uninstalling installer module...");
-
-        try {
-          if (!StringUtils.isBlank(postInstallFeatureStart)) {
-            featuresService.installFeature(
-                postInstallFeatureStart, EnumSet.of(FeaturesService.Option.NoAutoRefreshBundles));
-          }
-          if (!StringUtils.isBlank(postInstallFeatureStop)) {
-            featuresService.uninstallFeature(postInstallFeatureStop);
-          }
-        } catch (Exception e) {
-          LOGGER.debug("Error while trying to run the post-install start and stop operations.", e);
-        }
-
-      } else {
+
+      if (props.isEmpty()) {
         LOGGER.debug("No applications were found in the configuration file.");
+        return;
+      }
+
+      LOGGER.debug("Found applications to install from config.");
+      for (Entry<Object, Object> curApp : props.entrySet()) {
+        String appName = curApp.getKey().toString();
+        String appLocation = curApp.getValue().toString();
+        LOGGER.debug("Starting app {} at location: {}", appName, appLocation);
+
+        startApp(appName, appLocation);
       }
-      is.close();
+      LOGGER.debug("Finished installing applications, uninstalling installer module...");
+
+      installAndUninstallFeatures();
+
     } catch (FileNotFoundException fnfe) {
       LOGGER.warn("Could not file the configuration file at " + configFile.getAbsolutePath(), fnfe);
     } catch (IOException ioe) {
       LOGGER.warn("Could not load file as property list.", ioe);
-    } finally {
-      IOUtils.closeQuietly(is);
+    }
+  }
+
+  private void startApp(String appName, String appLocation) {
+    try {
+      if (StringUtils.isNotEmpty(appLocation)) {
```

Comment:
❗️ I don't believe `appLocation` can be `null` here as if it was we would have gotten a NullPointerException above so calling `appLocation.isEmpty()` should be good enough.